src/fluree/db/index.cljc

```diff
@@ -79,6 +103,24 @@
    :t 0
    :leftmost? true})
 
+(defn new-leaf
+  [network dbid cmp flakes]
+  (let [empty-set (flake/sorted-set-by cmp)]
+    (-> (empty-leaf network dbid cmp)
+        (assoc :flakes empty-set)
+        (add-flakes flakes))))
+
+(defn descendant?
```

Comment:
Can we add a docstring here to make clear which is the potential ancestor and which the potential descendant? Just to make it easier to use in the future when we've forgotten precisely which way the arguments were arranged : )